### PR TITLE
Fix Lens Flare GLES2

### PR DESCRIPTION
**Pull request by skhiat**
---
### Purpose of this PR
Fix regression on GLES2 only.
Move a function where HLSLcc will convert to available instruction.
https://fogbugz.unity3d.com/f/cases/1343061/
https://fogbugz.unity3d.com/f/cases/1343443/

---
### Testing status
Have to do a testing on GLES2 hardware.

### CODE REVIEW

File: com.unity.render-pipelines.core/Runtime/PostProcessing/Shaders/LensFlareCommon.hlsl [hunk 1]
```diff
@@ -96,7 +96,7 @@ float GetOcclusion(float2 screenPos, float flareDepth, float ratio)
 
     for (uint i = 0; i < (uint)_OcclusionSampleCount; i++)
     {
-        float2 dir = _OcclusionRadius * SampleDiskUniform(Hash(2 * i + 0 + 1), Hash(2 * i + 1 + 1));
+        float2 dir = _OcclusionRadius * SampleDiskUniform(Hash(2 * i + 0), Hash(2 * i + 1));
```
Comment (ellioman) [inline com.unity.render-pipelines.core/Runtime/PostProcessing/Shaders/LensFlareCommon.hlsl:99]: Any reason why we have `+ 0` there?

Comment (skhiat) [reply com.unity.render-pipelines.core/Runtime/PostProcessing/Shaders/LensFlareCommon.hlsl:99]: Yes, to make the code more readable, that have no perf impact.

**Comment by nigeljw-unity** [comment]
@Unity-Technologies/mobile-qa can you prioritize this as soon as possible, as it fixes a regression, so it is high priority. We need this tested against gles2 or webgl, as our CI does not have coverage for it.

File: com.unity.render-pipelines.core/Runtime/PostProcessing/Shaders/LensFlareCommon.hlsl (hunk 1 above)
Comment (ellioman) [reply com.unity.render-pipelines.core/Runtime/PostProcessing/Shaders/LensFlareCommon.hlsl:112]: uhhh, okay. Gonna have to disagree on the readability aspect here.

**Review by ellioman — APPROVED** [review]

**Review by phi-lira — APPROVED** [review]

**Comment by niklas-hoglund** [comment]
What's the status of this? I noticed the FogBugz case was marked resolved a month ago, but our builds are still failing and this PR hasn't been merged.

https://fogbugz.unity3d.com/f/cases/1343061

**Review by ellioman — APPROVED** [review]

**Review by nigeljw-unity — APPROVED** [review]

**Comment by nigeljw-unity** [comment]
@aleksandrasdzikia did you get a chance to test this?

**Review by sebastienlagarde — APPROVED** [review]

**Comment by sebastienlagarde** [comment]
poke @aleksandrasdzikia

**Comment by aleksandrasdzikia** [comment]
As talked on slack, I can not approve it yet because it has issues. We are waiting for a response from @skhiat

**Review by aleksandrasdzikia — APPROVED** [review]

**Review by phi-lira — APPROVED** [review]

**Comment by phi-lira** [comment]
PR is ready for approved but there was not ABV job kicked. I triggered it.

**Comment by phi-lira** [comment]
ABV failures either same as in trunk that would be solved by merging trunk

**Pull request merged**